options/rtdl/generic/linker.cpp

```diff
@@ -669,7 +680,11 @@ void ObjectRepository::_parseDynamic(SharedObject *object) {
 			object->preInitArraySize = dynamic->d_un.d_val;
 			break;
 		case DT_DEBUG:
+#if ELF_CLASS == 32
+			dynamic->d_un.d_val = reinterpret_cast<Elf32_Word>(&globalDebugInterface);
+#elif ELF_CLASS == 64
 			dynamic->d_un.d_val = reinterpret_cast<Elf64_Xword>(&globalDebugInterface);
+#endif
```

Comment:
Could this not use the `elf_` types? Or does this not work with the `using` types?

Reply:
I'm not sure how this exactly works, so I didn't define a type for this - it would only have this one use anyways.

Reply:
Even if it would only have this use, I feel like it would make the code more consistent. I'm however not familiar enough with C++ to say if it would work or not.